integrationTest/integration_test.go

```diff
@@ -725,57 +722,142 @@ func makeTests(t *testing.T) []*TestGroup {
 			tc("Add a new record - ignoring **.foo.com. targets", a("bar", "1.2.3.4"), ignoreTarget("**.foo.com.", "CNAME")),
 		),
 
-		testgroup("single TXT",
+		testgroup("simple TXT",
 			tc("Create a TXT", txt("foo", "simple")),
 			tc("Change a TXT", txt("foo", "changed")),
-			clear(),
 			tc("Create a TXT with spaces", txt("foo", "with spaces")),
-			tc("Create 1 TXT as array", txtmulti("foo", []string{"simple"})), // Same as non-TXTMulti
-			clear(),
-			tc("Create a 254-byte TXT", txt("foo", strings.Repeat("A", 254))),
 		),
 
-		testgroup("max-sized TXT",
-			not(
-				"INWX",  // INWX does not support
-				"MSDNS", // Supports 254-byte strings, not 255.  Seems like a bug.
-			),
-			tc("Create a 255-byte TXT", txt("foo", strings.Repeat("A", 255))),
+		testgroup("long TXT",
+			tc("Create long TXT", txt("foo", strings.Repeat("A", 300))),
+			tc("Change long TXT", txt("foo", strings.Repeat("B", 310))),
+			tc("Create long TXT with spaces", txt("foo", strings.Repeat("X", 200)+" "+strings.Repeat("Y", 200))),
 		),
 
-		testgroup("single TXT with single-quote",
-			not(
-				"INWX",    // Bug in the API prevents this.
-				"MSDNS",   // TODO(tlim): Should be easy to implement.
-				"CLOUDNS", // support txt("foo", "blah'blah") but does not support txt("foo","blah`blah")
-			),
-			tc("Create TXT with single-quote", txt("foo", "blah`blah")),
+		// In this next section we test all the edge cases related to TXT
+		// records. Compliance with the RFCs varies greatly with each provider.
+		// Rather than creating a "Capability" for each possible different
+		// failing or malcompliance (there would be many!), each provider
+		// supplies a function AuditRecords() which returns an error if
+		// the provider can not support a record.
+		// The integration tests use this feedback to skip tests that we know would fail.
+		// (Elsewhere the result of AuditRecords() is used in the
+		// "dnscontrol check" phase.)
+
+		testgroup("complex TXT",
+			// Do not use only()/not()/requires() in this section.
+			// If your provider needs to skip one of these tests, update
+			// "provider/*/recordaudit.AuditRecords()" to reject that kind
+			// of record. When the provider fixes the bug or changes behavior,
+			// update the AuditRecords().
+			tc("TXT with 0-octel string", txt("foo1", "")),
+			// https://github.com/StackExchange/dnscontrol/issues/598
+			// RFC1035 permits this, but rarely do provider support it.
+			clear(),
+			tc("Create a 253-byte TXT", txt("foo253", strings.Repeat("A", 254))),
```

Comment:
```suggestion
			tc("Create a 253-byte TXT", txt("foo253", strings.Repeat("A", 253))),
```

Thank you for adding these test cases, greatly appreciated for the HETZNER provider :)